async-nats/src/client.rs

```diff
@@ -90,7 +90,7 @@ impl Client {
             sender,
             next_subscription_id: Arc::new(AtomicU64::new(1)),
             subscription_capacity: capacity,
-            inbox_prefix,
+            inbox_prefix: Arc::new(inbox_prefix),
```

Comment:
```suggestion
            inbox_prefix: inbox_prefix.into(),
```
Is probably the way to go here